nmdc_runtime/site/changesheets/base.py

```diff
@@ -0,0 +1,80 @@
+# nmdc_runtime/site/changesheets/base.py
+"""
+base.py: Provides data classes for creating changesheets for NMDC database objects.
+"""
+
+import logging
+import time
+from dataclasses import dataclass
+from pathlib import Path
+import requests
+from typing import Any, ClassVar, Dict
+
+from nmdc_runtime.site.resources import RuntimeApiUserClient
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s %(levelname)s %(" "message)s"
+)
+
+JSON_OBJECT = Dict[str, Any]
+CHANGESHEETS_DIR = Path(__file__).parent.absolute().joinpath("changesheets_output")
+
+
+@dataclass
+class ChangesheetLineItem:
+    """
+    A line item in a changesheet
+    """
+
+    id: str
+    action: str
+    attribute: str
+    value: str
+
+    @property
+    def line(self) -> str:
+        return f"{self.id}\t{self.action}\t{self.attribute}\t{self.value}"
+
+
+@dataclass
+class Changesheet:
+    """
+    A changesheet
+    """
+
+    name: str
+    line_items: list = None
+    header: ClassVar[str] = "id\taction\tattribute\tvalue"
+    output_dir: Path = None
+
+    def __post_init__(self):
+        self.line_items = []
+        if self.output_dir is None:
+            self.output_dir = CHANGESHEETS_DIR
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+        self.output_filename_root: str = f"{self.name}-{time.strftime('%Y%m%d-%H%M%S')}"
+        self.output_filename: str = f"{self.output_filename_root}.tsv"
+        self.output_filepath: Path = self.output_dir.joinpath(self.output_filename)
+
+    def validate_changesheet(self, client: RuntimeApiUserClient) -> bool:
+        """
+        Validate the changesheet
+        :return: None
+        """
+        logging.info(f"Validating changesheet {self.output_filepath}")
+        resp = requests.post(
+            "https://api.microbiomedata.org/metadata/changesheets:validate",
```

Comment:
I'm wondering whether there may be a situation where the person running this code want to use a different base URL (e.g. the dev environment versus the prod environment; or a different environment entirely). I suggest making the base URL be dynamic (maybe basing it upon something in an environment variable, or at least a constant at the top of the module).

Reply:
updated validate_changesheet to require a base_url argument, taken from .env by the caller